vector/src/main/java/im/vector/app/core/utils/TextUtils.kt

```diff
@@ -86,49 +87,57 @@ object TextUtils {
     }
 
     fun formatDurationWithUnits(context: Context, duration: Duration): String {
+        return formatDurationWithUnits(duration, context::getString)
+    }
+
+    fun formatDurationWithUnits(stringProvider: StringProvider, duration: Duration): String {
+        return formatDurationWithUnits(duration, stringProvider::getString)
+    }
+
+    private fun formatDurationWithUnits(duration: Duration, getString: ((Int) -> String)): String {
```

Comment:
I would add  a doc/comment for this function to be clear what it does

Reply:
Nice idea, done.